theseus/optimizer/nonlinear/levenberg_marquardt.py

```diff
@@ -115,11 +116,17 @@ def compute_delta(
         damping_eps: Optional[float] = None,
         **kwargs,
     ) -> torch.Tensor:
-        if ellipsoidal_damping and not self._allows_ellipsoidal:
-            raise NotImplementedError(
-                f"Ellipsoidal damping is only supported by solvers with type "
-                f"[{_EDAMP_SOLVERS_STR}]."
-            )
+        if ellipsoidal_damping:
+            if not self._allows_ellipsoidal:
+                raise NotImplementedError(
+                    f"Ellipsoidal damping is only supported by solvers with type "
+                    f"[{_EDAMP_SOLVERS_STR}]."
+                )
+            if isinstance(self._damping, torch.Tensor):
```

Comment:
Do we need to disable `epplipsodial_damping` for `adaptive_damping=False`? Looks like the computation of `den` is incorrect if diagonal scaling is used.